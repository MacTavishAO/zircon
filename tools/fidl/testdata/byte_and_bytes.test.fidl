library example;

struct ByteAndBytes {
    byte single_byte;
    bytes many_bytes;
    bytes:1024 only_one_k_bytes;
    bytes:1024? opt_only_one_k_bytes;
};
