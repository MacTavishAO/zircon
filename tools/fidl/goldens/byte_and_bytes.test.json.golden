{
  "version": "0.0.1",
  "name": "example",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [],
  "service_declarations": [],
  "struct_declarations": [
    {
      "name": "example/ByteAndBytes",
      "location": {
        "filename": "byte_and_bytes.test.fidl",
        "line": 3,
        "column": 8,
        "length": 12
      },
      "anonymous": false,
      "members": [
        {
          "type": {
            "kind": "primitive",
            "subtype": "uint8"
          },
          "name": "single_byte",
          "location": {
            "filename": "byte_and_bytes.test.fidl",
            "line": 4,
            "column": 10,
            "length": 11
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 7
          }
        },
        {
          "type": {
            "kind": "vector",
            "element_type": {
              "kind": "primitive",
              "subtype": "uint8"
            },
            "nullable": false
          },
          "name": "many_bytes",
          "location": {
            "filename": "byte_and_bytes.test.fidl",
            "line": 5,
            "column": 11,
            "length": 10
          },
          "field_shape_v1": {
            "offset": 8,
            "padding": 0
          }
        },
        {
          "type": {
            "kind": "vector",
            "element_type": {
              "kind": "primitive",
              "subtype": "uint8"
            },
            "maybe_element_count": 1024,
            "nullable": false
          },
          "name": "only_one_k_bytes",
          "location": {
            "filename": "byte_and_bytes.test.fidl",
            "line": 6,
            "column": 16,
            "length": 16
          },
          "field_shape_v1": {
            "offset": 24,
            "padding": 0
          }
        },
        {
          "type": {
            "kind": "vector",
            "element_type": {
              "kind": "primitive",
              "subtype": "uint8"
            },
            "maybe_element_count": 1024,
            "nullable": true
          },
          "name": "opt_only_one_k_bytes",
          "location": {
            "filename": "byte_and_bytes.test.fidl",
            "line": 7,
            "column": 17,
            "length": 20
          },
          "field_shape_v1": {
            "offset": 40,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 56,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "example/ByteAndBytes"
  ],
  "declarations": {
    "example/ByteAndBytes": "struct"
  }
}
